src/PrestaShopBundle/Routing/Converter/AbstractLegacyRouteProvider.php

```diff
@@ -90,10 +90,28 @@ public function getLegacyRouteByAction($controller, $action)
             );
         }
 
+        $routeName = $this->getRouteName($controllerActions, $action);
+
+        return $this->getLegacyRoutes()[$routeName];
+    }
+
+    /**
+     * Get the route name.
+     * If there is more than one entry, return the last element.
+     *
+     * @param array $controllerActions
+     * @param string $action
+     *
+     * @return string
+     */
+    private function getRouteName(array $controllerActions, $action)
+    {
         $routeName = $controllerActions[$action];
-        $legacyRoutes = $this->getLegacyRoutes();
+        if (is_array($routeName)) {
+            return end($routeName);
```

Comment:
`_legacy_link` aren't stacked as Symfony routes are. 
If a module creates a route and the developer want to change all `AdminCategories` links with his own route, he'll use `_legacy_link: AdminCategories` too.
It becomes helpful to get the last one instead of Core behavior.